FEP-0008: Project Group Structure (#23)
FreeCAD/FreeCAD-Enhancement-Proposals: FEPs/FEP-0008-project-group-structure/README.md

Existing members can be removed from the group by the Chair if they do not longer fulfill the
requirements or the violate Code of Conduct [[2]](#code-of-conduct).

Since official communication takes place on GitHub, all members must have a GitHub account.
Membership is defined by the corresponding GitHub team: `@FreeCAD/<name>-working-group`.

Groups may define additional roles, but the structure should be kept as simple as possible.

### Communication Within a Group

Groups may choose their preferred communication tools, but the chosen platform must allow free
participation without barriers such as payment or special permissions. The method should be as
transparent as possible, ideally allowing read-only access without requiring an account.

Decisions (such as policy proposals, feature opinions, or issue feedback) must involve all group
members and, where appropriate, input from the wider community to avoid echo chambers. The Chair is
responsible for seeking compromise and consensus. If full consensus cannot be reached, the Chair
will summarize the prevailing opinion using the “rough consensus” rule [[1]](#rough-consensus).

> Any official group statement must be posted on GitHub (as a comment on a PR/issue, a new issue, or
> an FEP). Statements must clearly represent the group's position, not an individual opinion, and
> should include a summary of the discussion and a link to the full public conversation. Group statements that are not posted on GitHub shall not be cited as justification for any decision. Decisions must not be made concurrently with the publication of an official group statement. There must be a reasonable interval between an official group statement and any related decision, during which community members who are not part of the group can participate in the discussion.

Author: ```suggestion
Any official group statement must be posted on GitHub (as a comment on a PR/issue, a new issue, or
an FEP). Statements must clearly represent the group's position, not an individual opinion, and
should include a summary of the discussion. In case of group not being able to reach full consensus
over the matter, the summary must contain information about concerns raised within discussion.
Additionally, if possible a link to the full public conversation shall be posted along the summry. 

Group statements that are not posted on GitHub shall not be cited as justification for any decision.
Decisions must not be made concurrently with the publication of an official group statement. There
must be a reasonable interval between an official group statement and any related decision, during
which community members who are not part of the group can participate in the discussion.
```

interest, such as UI/UX, infrastructure, code quality, or CAD industry practices. Groups may focus
on development, infrastructure, policies, or other project-related topics.

The purpose of Working Groups is to provide a more organized venue for discussion and to ensure that such discussions yield useful outcomes. Each Working Group must conduct a genuine, documented discussion process. Working Groups must not be used solely to confer legitimacy on a predetermined decision.

Working Groups have no special decision-making authority unless explicitly delegated by the Maintainers; even in such cases, responsibility rests with the Maintainers, who grant and oversee that authority. Their primary function is to act as an advisory body. Working Groups shall not be held responsible for any decisions.

Working Group statements do not carry any special status over other community views. Maintainers should treat them as such when making decisions. If Maintainers cite a Working Group’s position as a rationale for a decision, the legitimacy of that decision derives from the quality and breadth of the group’s discussion—not from the mere existence of a Working Group statement.

> 

Reviewer: ```suggestion

At the Group’s request, the Chairperson, a Co-Chair, or a designated member may use GitHub features (e.g., labels, Projects, etc.) within the FreeCAD organization, under authorization from a Maintainer.

```

Author: That is not applied by the DWG members, but by me or Max based on the discussion. Labels are mostly for maintainers to keep everything under control. We indeed can allow chair to update them but that would need to be done in some automatic manner.

Reviewer: The issue here isn't about how to use these, but rather how to do so. In other words, any group can use them if they choose. What I mean is, there won't be any discrimination.

Author: Labels are project wide and displayed everywhere - they are for maintainers, not for groups to use. They of course can be added depending on actions of certain groups, but they are not _for_ groups. DWG Approved label was introduced not because the DWG wanted it but for communication between maintainers so it is easy to see what is already approved by DWG or not without needing to go through comments etc. I don't mind giving certain groups or certain people power to assign labels - but as this is project wide (boards are not, stuff that is in project board stays in the board and does not leak) that should be left for maintainers. Remember that any unwritten stuff is essentially for maintainers to decide, so if some group wants some label - they can always ask. The label should be however added because it brings value for project management and not simply because group wants it.

I also think that we should avoid mentioning groups in the labels and have independent structure like:

```
// For PRs
Requires: UI/UX review
Approved: UI/UX

// For Issues
Status: needs UI/UX assessment
Status: has UI/UX design

// for PRs where we don't only need normal review but more discussion over certain matter
Requires: Code Quality discussion

// for "business value"
Requires: CAD Review
Approved: CAD
```

Reviewer: As an outsider, this is how I interpret what's what. I believe the purpose of this document is to explain what these things are. As you can see, I misunderstood. Those less involved will misunderstand more. Please add an explanation of these to the document.

Reviewer 2: While you are correct that this document defines what things are, the use of labels is quite working group specific and can be defined for each working group instead.
I don't think we can completely define everything in this document, neither do I think we should. If we try to fully constrain how working group needs to operate, then there's a risk of over-constraining it.

It would be good if working groups has a document how they operate, maybe in the same place as their rules, but in my opinion, it doesn't need to be here.

to multiple Working Groups.

New members of the working group are appointed by the Chair. Community member may request membership by
contacting the Chair or asking over public channel. The Chair can accept or deny such request - in such
case however the Chair should provide rationale for the decision, for example lack of required expertise. Not being a member does not however forbid anyone from taking part in the discussions.

Existing members can be removed from the group by the Chair if they do not longer fulfill the
requirements or the violate Code of Conduct [[2]](#code-of-conduct).

Since official communication takes place on GitHub, all members must have a GitHub account.
Membership is defined by the corresponding GitHub team: `@FreeCAD/<name>-working-group`.

Groups may define additional roles, but the structure should be kept as simple as possible.

### Communication Within a Group

Groups may choose their preferred communication tools, but the chosen platform must allow free
participation without barriers such as payment or special permissions. The method should be as
transparent as possible, ideally allowing read-only access without requiring an account.

> Decisions (such as policy proposals, feature opinions, or issue feedback) must involve all group
> members and, where appropriate, input from the wider community to avoid echo chambers. The Chair is

Author: ```suggestion
Decisions (such as policy proposals, feature opinions, or issue feedback) must involve all group
members and, where appropriate, input from the wider community to avoid missing on potentially important feedback. The Chair is
```

Discord: https://discord.gg/PzYEBh5aA8

Channels:
- `#design-working-group` – general discussion
- `#ux-topics` – topic-specific discussions

#### Members

- @obelisk79 (chairperson)
- @kadet1090 (co-chair)
- @maxwxyz (co-chair)
- @pierreporte
- @MisterMakerNL
- @tigert

##### Membership Requirements

A member should:

* Participate actively in UI/UX discussions
* Seek consensus, even if it is not aligned with personal opinion
* Provide feedback in line with good UI/UX practices
* Research multiple solutions before providing feedback

The group seeks members with strong UI/UX intuition or experience. Being a
developer is not required but is an advantage.

### CAD Working Group (CWG)

The main goal of CWG is to ensure that FreeCAD is developed in a way that makes it possible to use it
in professional scenarios, aligning it with industry standards and ensuring that most daring issues
are tackled first.

#### Tasks of the group
 * Establishing CAD development roadmap
 * Providing feedback for new features

### Communication

Description of how to participate in discussions with the group, with links if needed.

#### Members
 * @pierreporte (chair)

##### Membership Requirements

> _(TBD)_

Reviewer 3: LGTM

coherent and maintainable way; final decisions rest with them.

However, it is impossible for a single person to have deep expertise in every area of the project.
To ensure that all areas are handled well and that development remains consistent, Maintainers often
seek advice from **Working Groups**. While their recommendations are usually followed, Maintainers
may also consider other factors, such as community impact or maintenance costs, before making a
final decision.

### Current Project Maintainers

In alphabetical order:

- Adrián Insaurralde Avalos (@adrianinsaval)
- Benjamin Nauck (@hyarion)
- **Brad Collette** (@sliptonic)
- **Chris Hennes** (@chennes)
- Kacper Donat (@kadet1090)
- @Roy-043
- @WandererFan
- **Werner Mayer** (@wwmayer)
- **Yorik van Havre** (@yorik)

**Bolded** names indicate Project Administrators.

#### Inactive Maintainers

Some Maintainers who were once active are no longer participating in development. While they do not
lose their Maintainer status, those who are inactive for an extended period will have their write
access temporarily suspended for security reasons. Access is restored if and when they return.

Current list of inactive maintainers:

> - ...

Reviewer 3: ```suggestion
The current project Maintainers team can be viewed on the GitHub organization: https://github.com/orgs/FreeCAD/teams/maintainers
```

If needed, admins can be added manually (I assume they do not change regularly)

Only the Chairperson or a Co-Chair may speak on behalf of the Group. If a member makes a statement purporting to represent the Group, confirmation may be requested from the Chairperson or the Co-Chairs. Unless it is expressly stated that the speaker is acting on behalf of the Group, all statements are deemed to be made in a personal capacity.

### Creating a New Working Group

Because Working Groups are permanent, their creation requires extended discussion. A new Working
Group must not overlap or conflict with existing groups and should address an area that is currently
uncovered.

Working Groups are created by submitting a new FEP establishing the group with motivation and rationale. The PR introducing new group should also update this FEP with specification of the new group using template below:

```
### Name of the group (Abbr)

Short description of the group.

#### Tasks of the group
 * Task1
 * Task2
 * ...

#### Non-Tasks of the group
 * Task1
 * Task2
 * ...

#### Communication

Description of how to participate in discussions with the group, with links if needed.

#### Members
 * @chairperson (chairperson)
 * @cochair1 (co-chair)
 * @member1

>  * @member2

Reviewer 3: ```suggestion
 * Link to the GitHub team
```

Author: shouldn't this replace the list?

Reviewer 3: I guess, optional

Reviewer 3: @kadet1090 adjusted

This section lists currently active Working Groups.

### Design Working Group (DWG)

The DWG provides advice and feedback on user interface changes and consistency. It works with
developers and Maintainers to propose solutions that improve the coherence and usability of
FreeCAD’s user experience across all platforms.

#### Tasks of the group

- Providing feedback on interface changes and consistency
- Creating guidelines to improve FreeCAD UI/UX
- Ensuring UI/UX consistency within FreeCAD
- Ensuring UI/UX consistency with other applications and CAD suites
- Helping design interfaces for new features
- Maintaining the UX/UI roadmap

#### Non-Tasks of the group

The group consists mainly of UI/UX experts, not developers, so certain tasks are out of scope:

- Implementing suggestions in PRs
- Providing advice on implementation details

#### Communication

Discord: https://discord.gg/PzYEBh5aA8

Channels:
- `#design-working-group` -- general discussion
- `#ux-discussions` -- specific discussions
- `#ux-reviews` -- review-specific discussions

#### Members

- @obelisk79 (chairperson)
- @kadet1090 (co-chair)
- @maxwxyz (co-chair)
- @pierreporte
- @MisterMakerNL

> - @tigert

Reviewer 3: ```suggestion
The list of current Design Working Group members can be viewed here: https://github.com/orgs/FreeCAD/teams/design-working-group
```

If needed, chair and other positions can be added manually if they won't change frequently.

Due to FreeCAD nature, a part of the CWG is more inclined towards mechanical engineering aspects while an other part is more interested in civil engineering and architecture.

#### Tasks of the group

* Providing feedback for feature changes and new features
* Ensuring industrial standards are followed
* Ensuring that industrial good practices are followed
* Research missing features
* Research strength and flaws to existing solution to make FreeCAD solutions better
* Establishing and maintaining CAD development roadmap

#### Non-Tasks of the group

The group consists mainly in CAD professionals, not developers, so certain tasks are out of scope:

* Implementing suggestions in PRs
* Providing advice on implementation details

Members are free to perform these tasks but they would not count as CWG activity.

#### Communication

Discord: https://discord.gg/PzYEBh5aA8

Channels:

* `#cad-working-group` -- general discussion
* `#cad-topics` -- topic-specific-discussions

#### Members

* @pierreporte (chair)
* @maxwxyz (co-chair)
* @yorikvanhavre
* @kaiwas
* @paullee0
* @marcuspollio
* @furgo16
* @semhustej
* @FEA-Eng

> * @tigert

Reviewer 3: ```suggestion
The list of current CAD Working Group members can be viewed here: https://github.com/orgs/FreeCAD/teams/cad-working-group
```

##### Membership Requirements

A member should:

* Participate actively in CAD features discussions
* Seek consensus, even if it son aligned with personal opinion
* Provide feedback in line with good industry practices
* Research multiple solutions before providing feedback

The group seeks member with strong CAD experience. Working as an engineer, draftsman or architect is not required but strongly prefered.

### Code Quality Working Group (CQWG)

The CQWG provides advice and feedback on code quality, establishes best practices, and helps
contributors write clean and maintainable code.

#### Tasks of the group

- Reviewing PRs for code quality
- Establishing best practices for the FreeCAD codebase
- Creating guidelines for code reviewers
- Providing advice to contributors on code quality

#### Communication

Discord: https://discord.gg/hDh4zxhkrw

Channels:
- `#code-quality-working-group` – general discussion
- `#code-quality-topics` – topic-specific discussions

#### Members

- @kadet1090 (chairperson)
- @hyarion (co-chair)
- @bensay
- @tritao
- @pieterhijma

> - @captain0xff

Reviewer 3: ```suggestion
The list of current Code Quality Working Group members can be viewed here: https://github.com/orgs/FreeCAD/teams/code-quality-working-group
```
